crates/sui-rpc-api/src/lib.rs

```diff
@@ -80,7 +80,9 @@ impl RpcService {
         let metrics = self.metrics.clone();
 
         let mut router = {
-            let node_service = crate::proto::node::node_server::NodeServer::new(self.clone());
+            let node_service = crate::proto::node::node_server::NodeServer::new(self.clone())
+                // Set the maximum message size to 10MB as some checkpoints are over 4 MB in size
+                .max_decoding_message_size(10 * 1024 * 1024);
```

Comment:
This needs to be set by the client and not the server. And given the maximum size of a checkpoint is ~32MB (sans objects), lets set the maximum decode size of the fullcheckpoint api to 64MB to give us some more headroom